### PR TITLE
Pimpl for Queries

### PR DESCRIPTION
**Pull request by Akvinikym**
### Description of the Change

This PR inroduces Pimpl idiom for Queries.

### Benefits

Faster build.

### Possible Drawbacks 

None.

### CODE REVIEW

File: shared_model/backend/protobuf/queries/impl/proto_query.cpp [hunk 1]
```diff
@@ -4,92 +4,140 @@
...
-
-    template Query::Query(Query::TransportType &);
-    template Query::Query(const Query::TransportType &);
-    template Query::Query(Query::TransportType &&);
-
-    Query::Query(const Query &o) : Query(o.proto_) {}
-
-    Query::Query(Query &&o) noexcept : Query(std::move(o.proto_)) {}
+    struct Query::Impl {
+      explicit Impl(TransportType &&ref) : proto_{std::move(ref)} {}
```
Comment (igor-egorov) [inline shared_model/backend/protobuf/queries/impl/proto_query.cpp]: test

Comment (igor-egorov) [inline shared_model/backend/protobuf/queries/impl/proto_query.cpp]: test

File: shared_model/backend/protobuf/queries/impl/proto_query.cpp [hunk 2]
```diff
@@ -4,92 +4,140 @@
...
+                        .GetDescriptor()
+                        ->FindFieldByNumber(ar.payload().query_case())
+                        ->index_in_oneof();
+        return shared_model::detail::variant_impl<ProtoQueryListType>::
+            template load<ProtoQueryVariantType>(std::forward<decltype(ar)>(ar),
+                                                 which);
+      }()};
+
+      QueryVariantType ivariant_{
+          [this] { return QueryVariantType(variant_); }()};
```
Comment (MBoldyrev) [inline shared_model/backend/protobuf/queries/impl/proto_query.cpp]: I understand you did not add this code in this PR, but the lambda seems useless here, if we can just write
`QueryVariantType ivariant_{QueryVariantType(variant_)};`

File: shared_model/backend/protobuf/queries/impl/proto_query.cpp [hunk 3]
```diff
@@ -4,92 +4,140 @@
...
+
+      interface::types::BlobType blob_{[this] { return makeBlob(*proto_); }()};
```
Comment (MBoldyrev) [inline shared_model/backend/protobuf/queries/impl/proto_query.cpp]:
```suggestion
      interface::types::BlobType blob_{makeBlob(*proto_)};
```

File: shared_model/backend/protobuf/queries/impl/proto_query.cpp [hunk 4]
```diff
@@ -4,92 +4,140 @@
...
+
+      interface::types::BlobType payload_{
+          [this] { return makeBlob(proto_->payload()); }()};
```
Comment (MBoldyrev) [inline shared_model/backend/protobuf/queries/impl/proto_query.cpp]: same about lambda

File: shared_model/backend/protobuf/queries/impl/proto_query.cpp (hunk 2 above)
Comment (lebdron) [reply shared_model/backend/protobuf/queries/impl/proto_query.cpp]: Maybe even simpler in this case
`QueryVariantType ivariant_{variant_};`

Comment (MBoldyrev) [reply shared_model/backend/protobuf/queries/impl/proto_query.cpp]: yes)

**Pull request merged**